Add `html` component

wwwillchen opened the pull request:
This allows you to render sanitized HTML, however it doesn't allow potentially dangerous HTML like JavaScript.

In a separate PR, I'll update the `embed` component so you can essentially do an `<iframe srcdoc="$htmlcontent">`, which will allow you to run arbitrary JS in a sandboxed way.

Partially addresses #156.

richard-to commented on docs/components/html.md:
```diff
@@ -0,0 +1,17 @@
+
+## Examples
+
+<iframe class="component-demo" src="https://mesop-y677hytkra-uc.a.run.app/link"></iframe>
```
[nit] Looks like these are pointing to "link" when it should be html?

richard-to commented:
> 1. There's no styling inside the iframe. Should we inject the stylesheet in here?

What do you mean by "inject the stylesheet"?

When using the iframe with source doc, is the user able to reference assets (Javascript, CSS) from other domains or would they need to reference from same domain? Or is that not allowed for security reasons?

I guess if linking to external resources is not allowed, then one could write the js/css in the html head. Probably not that user friendly, but since this is an advanced use case, probably ok. Although would be hard to manage.

> 2. The iframe height is arbitrary (browser default). Should we try to automatically resize the iframe based on the content or should Mesop app developers set a height?

I think the app developer should able to set the height. So if there is auto resize as a default and the user could resize as an override, that may be more convenient.

richard-to reviewed (approved):
LGTM

wwwillchen commented:
will re-open once I've been able to address the comments.

wwwillchen commented:
@richard-to could you take another look? I updated this PR quite a bit and switched to sanitizing the HTML instead of iframing the HTML (I'll try this approach in a separate https://github.com/google/mesop/pull/405). This is better for some use cases because it gets rendered in the same frame, but it doesn't support JS so it's not that flexible.

richard-to commented on mesop/components/html/html.ts:
```diff
@@ -0,0 +1,51 @@
+    this._config = HtmlType.deserializeBinary(
+      this.type.getValue() as unknown as Uint8Array,
+    );
+    // const previousSrcDoc = this.srcDoc;
```
Can this commented out code be removed?

wwwillchen replied:
oops, I was too hasty - thanks, done.

richard-to commented on mesop/components/html/html.ts:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  ngAfterViewInit() {
+    // this.loadIframe();
```
Ditto here

richard-to commented on mesop/web/src/safe_iframe/safe_iframe.ts:
```diff
@@ -30,4 +36,24 @@ function setIframeSrcImpl(iframe: HTMLIFrameElement, src: string) {
+    'allow-storage-access-by-user-activation',
+  );
+
+  // Check if there's any santiziation that's needed.
```
nit: typo sanitization

wwwillchen replied:
(I'll add this file in #405

richard-to commented on mesop/components/html/html.ng.html:
```diff
@@ -0,0 +1 @@
+<div [innerHTML]="config().getHtml()" [style]="getStyle()"></div>
```
So this is no longer rendered in an iframe? I see some iframe changes in this PR still. How are those related?

wwwillchen replied:
Yeah, no longer rendered in an iframe. The iframe changes were left over by accident - I've removed them.

richard-to reviewed (approved):
LGTM - Just to make sure I understand. The HTML component here renders HTML directly without an iframe. But it is sanitized so no javascript and other malicious stuff can be added?

wwwillchen commented:
> LGTM - Just to make sure I understand. The HTML component here renders HTML directly without an iframe. But it is sanitized so no javascript and other malicious stuff can be added?

Yup, that's right. Right now we've been telling people to use me.markdown as a way to render sanitized HTML, which is 1) kind of confusing (since it's not obvious to most developers) and 2) kind of annoying since markdown adds a `<p>` which adds margin, and having a raw div makes it more straightforward to style.

I think we still need JS for a bunch of use cases, so we'll need to handle that separately.
